### PR TITLE
Filter

### LINKED ISSUE #47: Filteroptions
## User story
1. As a business manager
2. I want / need to adapt the visualizations to my needs without writing coded queries
3. So that I can quickly access and recognize the answers to my questions

## Acceptance criteria
* Filter nodes 
          - hierarchical structure as multi-select checkboxes
          - left to right indentation -> higher granularity 
* Max edges configurable (250 for now)
* Filter relations
         - multi-select checkboxes 
 * Foldable menu ->Exemplary visualization:
image
!! Search bar (placeholder for now)!!

* When checkboxes are clicked -> change visualization accordingly 
* Implemented for Dashboard #45 and clickable subpages with zoomed in visualizations 
* Displayed as button
* Text: "Apply filter"
* When button is clicked -> apply chosen filters to visualization(s)
* Exemplary visualization:
Bildschirmfoto 2021-05-19 um 08 14 36

## Definition of done
* Documentation conform with tsdoc
* Reviewed with 4 eyes principle
* Understandable documentation for developers

### PR DESCRIPTION
**Pull request by CatoLeanTruetschel**
This PR ultimately implements the filter options as described by #47. 

This is not true any longer due to our new branching strategy.

### CODE REVIEW

File: backend/src/filter/filter.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  EdgeTypeFilterModel,
+  FilterModelEntry,
+  NodeTypeFilterModel,
+} from '../shared/filter';
+
+@Injectable()
+export class FilterService implements FilterServiceBase {
+  constructor(private readonly neo4jService: Neo4jService) {}
+
+  public async getNodeTypeFilterModel(
```
Comment (joluj) [inline backend/src/filter/filter.service.ts:14]: comment missing

File: backend/src/filter/filter.service.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      (x) => x.toObject() as FilterModelEntry
+    );
+
+    return {
+      name: type,
+      properties,
+    };
+  }
+
+  public async getEdgeTypeFilterModel(
```
Comment (joluj) [inline backend/src/filter/filter.service.ts:44]: comment

File: backend/tests/fixtures/filter/FilterServicMock.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  EdgeTypeFilterModel,
+  NodeTypeFilterModel,
+} from '../../../src/shared/filter';
+import {
+  getEdgeTypeFilterModelResult,
+  getNodeTypeFilterModelResult,
+} from './FilterQueryResults';
+
+export default class FilterServiceMock implements FilterServiceBase {
+  // eslint-disable-next-line class-methods-use-this
```
Comment (joluj) [inline backend/tests/fixtures/filter/FilterServicMock.ts:12]: disable this globally as described in some other pull request as well

**Review by joluj — APPROVED** [review]
Comments for the service is missing. Since it's 1:1 the controller it's ok

File: backend/src/filter/filter.service.ts (hunk 1 above)
Comment (CatoLeanTruetschel) [reply backend/src/filter/filter.service.ts:14]: I opened #141 for this and will implement this in a follow up.

File: backend/src/filter/filter.service.ts (hunk 2 above)
Comment (CatoLeanTruetschel) [reply backend/src/filter/filter.service.ts:44]: I opened #141 for this and will implement this in a follow up.

File: backend/tests/fixtures/filter/FilterServicMock.ts (hunk 3 above)
Comment (CatoLeanTruetschel) [reply backend/tests/fixtures/filter/FilterServicMock.ts:12]: I openend  #142 to track this.

**Pull request merged**